docs/assets/diagrams/transfer/02-create-transaction.puml

```diff
@@ -0,0 +1,124 @@
+@startuml
+
+title Create Transaction
+
+hide footbox
+
+autonumber
+
+box "Mobile Device" #Light
+  actor Alice
+  participant "PISP App" as App
+end box
+box "PISP" #Light
+  database Firebase
+  participant "PISP Server" as Server
+end box
+box "Mojaloop" #Light
+  participant "Third-party\nAPI Adapter" as Adapter
+end box
+
+Alice -> App: I want to make this transaction
+
+rnote right of App #Light 
+  ""createDoc(""
```

Comment:
To mention my previous discussion with @looilee, he suggested to separate the lookup to decouple it from the transaction. So perhaps the app can also listen to `lookups/users/1234` which stores all the result for party lookups performed and delete it once it is unused (cmiiw). That way, `createDoc("transactions/users/1234/<uuid>", ...)` will be performed only after the user confirms the payee and specify the transfer amount. In my opinion, it is also a good way to go since we may need to perform party lookup in the linking process as well. May I ask what do you think about this @jgeewax?

Reply:
In my recent update, I try to split the collections for transaction and lookup

Reply:
Reverted to the original flow.